modules/SitecoreImageBuilder/SitecoreImageBuilder.psm1

```diff
@@ -1,24 +1,37 @@
 function Invoke-Build
 {
-    [CmdletBinding(SupportsShouldProcess = $true)]
+    [CmdletBinding(SupportsShouldProcess = $true, DefaultParameterSetName = "__SourcePath")]
     param(        
         [Parameter(Mandatory = $true)]
         [ValidateScript( { Test-Path $_ -PathType 'Container' })] 
         [string]$Path,
-        [Parameter(Mandatory = $true)]
+        [Parameter(Mandatory = $true, ParameterSetName = "__SourcePath")]
         [ValidateScript( { Test-Path $_ -PathType 'Container' })] 
         [string]$InstallSourcePath,
+        [Parameter(Mandatory = $true, ParameterSetName = "__SourceResolver")]
+        [ValidateScript({ $null -ne $_ })]
+        [scriptblock]$InstallSourceResolver,
         [Parameter(Mandatory = $true)]
         [string]$Registry,
         [Parameter(Mandatory = $false)]
-        [array]$Tags = @("*"),
+        [Alias("Tags")]
+        [string[]]$Include = @("*"),
+        [Parameter(Mandatory = $false)]
+        [string[]]$Exclude = @(),
         [Parameter(Mandatory = $false)]
         [ValidateSet("WhenChanged", "Always", "Never")]
         [string]$PushMode = "WhenChanged",
         [Parameter(Mandatory = $false)]
         [ValidateSet("Always", "Never")]
         [string]$PullMode = "Always"
     )
+
+    If ($PSCmdlet.ParameterSetName -eq "SourceFromPath") {
```

Comment:
`SourceFromPath` should be `__SourcePath` else `$InstallSourceResolver` will be null and blow up later

Reply:
Oops, artifact of refactor. Fix pending (good catch)